### PR TITLE
Add PHPStan (config) and some fixes

### PR DESCRIPTION
**Pull request by tersmitten**

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -0,0 +1,4 @@
+parameters:
+    ignoreErrors:
+#        - '#Parameter \$to of method Slim\\App::redirect\(\) has invalid typehint type Slim\\UriInterface#'
+#        - '#Variable \$_FILES in isset\(\) always exists and is not nullable#'
```
Comment (tersmitten) [inline phpstan.neon:4]: Some example config, to show how to ignore certain (found) errors

File: .travis.yml [hunk 2]
```diff
@@ -13,12 +13,14 @@ php:
...
 before_script:
   - if [[ "$TRAVIS_PHP_VERSION" == '5.6' ]]; then composer require satooshi/php-coveralls:1.* squizlabs/php_codesniffer:2.* -n ; fi
+  - if [[ "$TRAVIS_PHP_VERSION" == '7.1' ]]; then composer require phpstan/phpstan:^0.10.3 -n ; fi
   - if [[ "$TRAVIS_PHP_VERSION" != '5.6' ]]; then composer install -n ; fi
 
 script:
   - if [[ "$TRAVIS_PHP_VERSION" == '5.6' ]]; then vendor/bin/phpunit --coverage-clover clover.xml ; fi
   - if [[ "$TRAVIS_PHP_VERSION" != '5.6' ]]; then vendor/bin/phpunit ; fi
   - if [[ "$TRAVIS_PHP_VERSION" == '5.6' ]]; then vendor/bin/phpcs ; fi
+  - if [[ "$TRAVIS_PHP_VERSION" == '7.1' ]]; then vendor/bin/phpstan analyse Slim -l 1 || true ; fi
```
Comment (tersmitten) [inline .travis.yml:23]: Allow failure

Comment (tersmitten) [inline .travis.yml:23]: It's only at level 1 now, this can be increased over the time

**Comment by akrabat** [comment]
Thanks!

**Pull request merged**